---
sidebarDepth: 3
sidebar: auto
prev:
  text: Back To 目录
  link: /java/
typora-root-url: ..\.vuepress\public
---



## 程序中异常产生后，是如何处理的

程序中一旦产生异常，首先会中断向下执行。如果没有处理，默认是将异常传递给本方法的调用者。不断往回传递，直到JVM收到该异常，此时程序终止执行。

## 异常体系

![image-20230315143427942](/images/springsecurity/image-20230315143427942.png)

- Error：严重问题，通过代码无法处理（比如内存溢出）
- Exception：称为异常类，它表示程序本身可以处理的问题
  - RuntimeException及其子类：运行时异常。 (空指针异常，数组索引越界异常) 
  - 非RuntimeException及其子类：编译时异常 , 编译期必须处理的，否则程序不能通过编译。 (日期格式化异常)。 

```java
public static void main(String[] args) {
    try{
        throw new RuntimeException("抛出一个Runtime异常");
    }catch (Exception e){
        System.out.println(e.getMessage());
    }
    System.out.println("main运行结束");
}
/**
     * 抛出一个Runtime异常
     * main运行结束
     */
```





## JVM默认处理异常的方式

把异常的类型 , 原因 , 位置打印在控制台，程序停止执行

**注意 : 程序中出现了异常 , 会在当前位置创建此异常的对象 , 对象中包含了异常的信息 , 并把此异常交给本方法的调用者处理**



